Plugin/Catalog/Block/DefaultParameterResolver.php

```diff
@@ -84,7 +84,7 @@ public function getCurrentPage()
     }
 
     /**
-     * @return string
+     * @return string|null
```

Comment:
I wouldn't expect this method to return null. Throw an exception instead.

Reply:
It returns null when doing a text search, as in this case the results are not related to one specific category.

Reply:
Gotcha